### ISSUE TITLE
Attempt to launch Android emulator on Windows failed with improperly trimmed `emulatorName`

### ISSUE THREAD

**Issue by qiushihe** [react-native-community/cli#1084]
## Environment

```
System:
    OS: Windows 10 10.0.18363
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 19.06 GB / 31.88 GB
  Binaries:
    Node: 12.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  Languages:
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0
    react-native: 0.62.0 => 0.62.0
  npmGlobalPackages:
    *react-native*: Not Found
```

## Description

On Windows, the `emulatorName` used by [tryLaunchEmulator.ts](https://github.com/react-native-community/cli/blob/master/packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts) is not properly trimmed off its trailing `\r`, thus causing the attempt to launch Android emulator to fail.

The `getEmulators` function here: https://github.com/react-native-community/cli/blob/master/packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts#L11 splits the output of `emulator -list-avds` by the newline character only which means, when on Windows the results from the split have a trailing `\r` character. This in turn cause the `execa` call in `launchEmulator` to fail.

## Reproducible Demo

To reproduce, you'll need to be on Windows.

For my example, the name of the AVD is `Android_API_28_-_Pixel_3_XL` and I can see it is the first result in the output of `emulator -list-avds`:

```
> C:\Users\blah\blah\blah\Android\Sdk\emulator\emulator.exe -list-avds
Android_API_28_-_Pixel_3_XL
Fossil_Gen_5_API_26
```

And to confirm my suspicion of the issue, I manually edited the `tryLaunchEmulator.js` (located at `my-project/node_modules/@react-native-community/cli-platform-android/build/commands/runAndroid/tryLaunchEmulator.js`) and inserted some logging above the `execa` call:

```
const launchEmulator = async (emulatorName: string, adbPath: string) => {
  return new Promise((resolve, reject) => {
    const avd1 = `@${emulatorName}`;
    const avd2 = "@Android_API_28_-_Pixel_3_XL";

    console.log("avd1", avd1);
    console.log(JSON.stringify(avd1));
    console.log(typeof avd1);
    console.log(avd1.length);

    console.log("avd2", avd2);
    console.log(JSON.stringify(avd2));
    console.log(typeof avd2);
    console.log(avd2.length);

    console.log("avd1 == avd2", avd1 == avd2);
    console.log("avd1 === avd2", avd1 === avd2);

    const cp = execa(emulatorCommand, [`@${emulatorName}`], {
      detached: true,
      stdio: 'ignore',
    });

    ...
  });
};
```

... and the output from the above shows:

```
info Launching emulator...
avd1 @Android_API_28_-_Pixel_3_XL
"@Android_API_28_-_Pixel_3_XL\r"
string
29
avd2 @Android_API_28_-_Pixel_3_XL
"@Android_API_28_-_Pixel_3_XL"
string
28
avd1 == avd2 false
avd1 === avd2 false
```

... notice the JSON stringified string of `avd1` has an extra `\r` on the end and that its `length` is `29` instead of `28`.

## Suggestion to Fix the Issue

I was able to fix the issue locally by either:

* Change [line 11](https://github.com/react-native-community/cli/blob/master/packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts#L11) by changing `emulatorsOutput.split('\n').filter(name => name !== '')` to ``emulatorsOutput.split('\n').filter(name => name !== '').map(name => `${name}`.trim())``; OR ...
* Change [line 19](https://github.com/react-native-community/cli/blob/master/packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts#L19) by changing ``[`@${emulatorName}`]`` to ``[`@${emulatorName}`.trim()]``

---

So in conclusion, this is not a huge blocker for me because I can still manually run the emulator before running `react-native run-android`. However if someone can take a look at this issue and get the fix implemented, that would be great. Thanks!

**Comment by qiushihe** [comment]
Also a note about the `react-native info` output above ☝️ . I have no idea why it says so many things are "Not found" because they are definitely installed (except `yarn`) and I am able to successfully run the emulator (manually) and my ReactNative app without problem 🤷‍♂ .

**Comment by thymikee** [comment]
This sounds like an issue with Android emulator, but if we can fix it with trimming the name, please do! Can you send a PR?

**Comment by Esemesek** [comment]
I think I found a better way to fix this issue. Created a PR: https://github.com/react-native-community/cli/pull/1098.
@qiushihe Can you please verify if it's working for you?

**Comment by qiushihe** [comment]
Hey @Esemesek looks like it would work (I tried it quickly in `node` console under Powershell) and it's a lot cleaner than what I hastily proposed above 😅 thanks!
